src/editor/Editor.js

```diff
@@ -279,9 +281,14 @@ define(function (require, exports, module) {
      * @param {{startLine: number, endLine: number}=} range If specified, range of lines within the document
      *          to display in this editor. Inclusive.
      */
-    function Editor(document, makeMasterEditor, container, range) {
+    function Editor(document, makeMasterEditor, container, range, options) {
         var self = this;
-        
+
+        var isReadOnly = false;
+        if (options !== undefined && typeof (options.isReadOnly) === "boolean") {
+            isReadOnly = options.isReadOnly;
+        }
```

Comment:
Yeah I think a stringent type check is not needed here. Did you mean the above condition to be `var isReadOnly = (options && options.isReadOnly)`